picard/coverart/image.py

```diff
@@ -216,6 +219,8 @@ def set_data(self, data):
 
         try:
             self.datahash = DataHash(data, suffix=self.extension)
+            if not self.original_datahash:
+                self.original_datahash = DataHash(data, suffix=self.extension)
```

Comment:
Hmmm actually a bad idea, it will recalculate the same stuff for nothing.
See https://github.com/metabrainz/picard/blob/master/picard/coverart/image.py#L56

Imho`original_data` and its hash should only be set and calculated when needed, which is on first processor run.

Imho we should limit to the bare minumum:
- if a processor has to run on the image
- test if image's `original_datahash` is None, set it to current hash
- copy the Datahash object (copying hash & linked file)
- run processor and use set_data() to update the copied object (if it isn't modified it will not create a new file)
- store `original_datahash` in new object

Basically we have cases where processors will do nothing, we shouldn't create more objects or files in this case.

Instead of providing test plugins as gist, create actual tests. Each hook should use a different processing (ie. different sizes).